tests/python/unittest/test_ir_container.py

```diff
@@ -34,6 +34,21 @@ def test_array_save_load_json():
     assert a_loaded[1].value == 2
 
 
+def test_dir_array():
+    a = tvm.runtime.convert([1, 2, 3])
+    dir(a)
+
+
+def test_getattr_array():
+    a = tvm.runtime.convert([1, 2, 3])
+    type_key = getattr(a, "type_key")
+    assert type_key == "Array"
+    try:
+        getattr(a, "test_key")
+    except AttributeError:
+        pass
```

Comment:
If you are expecting an exception, please use
```
with pytest.raise(AttributeError):
    getattr(a, "test_key")
```

In this particular case, it might be even simpler :

```
assert not hasattr(a, "test_key")
```